packages/next/cli/next-dev.ts

```diff
@@ -75,9 +75,20 @@ const nextDev: cliCommand = (argv) => {
       )
     }
   }
-  const allowRetry = !args['--port']
-  let port: number =
-    args['--port'] || (process.env.PORT && parseInt(process.env.PORT)) || 3000
+
+  const hasPortFlag = Object.hasOwnProperty.call(args, '--port')
+  const allowRetry = !hasPortFlag
+  let port: number
+  if (hasPortFlag) {
+    port = args['--port']
+  } else {
+    const parsed = process.env.PORT && parseInt(process.env.PORT)
+    if (typeof parsed === 'number' && !Number.isNaN(parsed)) {
+      port = parsed
+    } else {
+      port = 3000
+    }
+  }
```

Comment:
Should we move this to a lib file so we don't need to update it for both `next-dev` and `next-start`?